src/commcare_cloud/environment/secrets/secrets.yml

```diff
@@ -0,0 +1,168 @@
+- name: AIRFLOW_FLASK_KEY
```

Comment:
Is it necessary to add all new secret names to this file to have them be accessible by commcare-cloud, or do reasonable defaults kick in for new secrets that are not listed in here? If new secrets must be added to this file, is that obvious (in the error message, for example) if this step is missed/forgotten?

In case it wasn't obvious from the question, reasonable defaults would be nice.

Reply:
That's really good feedback and something that slipped my mind to address. The way it works right now is somewhat analogous to how it works before: previously you could add any variable you wanted to vault.yml and it would be available in ansible contexts, but if you wanted to check for the value and assign it a default if it's missing, then you had to add a line to `all.yml`; now a line in this file plays the role of a line in `all.yml`.

Now, once we're using a different secrets backend, for a secret to be accessible from either backend it will need to go in this file. That's because another backend might fetch secrets in ansible (the one I'm considering will use https://docs.ansible.com/ansible/latest/plugins/lookup/aws_secret.html for example) and so there are no secrets passed in directly as ansible variables using `-e` as is done with vault.yml currently.

I considered maybe having the ansible-vault backend print out info messages saying

> your vault file contains an unknown secret "MY_CUSTOM_SECRET". Consider removing it if it's not needed or adding it to `src/commcare_cloud/environment/secrets/secrets.yml` if it is

or doing some kind of error checking like that, but I didn't want to do something too intrusive/annoying in the first pass. For now I'll quickly document this and then keep a look out for convenient places in the workflow to link people to the documentation.

Reply:
This is hopefully a good start https://github.com/dimagi/commcare-cloud/pull/4122, though docs are one thing and getting people to actually know to **read** them is another. I still intend to sneak a link to that in somewhere that'll be visible at the moment that it'll be useful to someone (like when a new undocumented secret is detected or something)